packages/pwa-kit-extension-sdk/src/configs/webpack/override-stats-plugin.test.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {runWebpackCompiler} from './test-utils'
+import OverrideStatsPlugin, {OverrideStatsEntry} from './override-stats-plugin'
+import {Compiler, Compilation} from 'webpack'
+
+// Custom plugin to set overrideStats during compilation
+class SetOverrideStatsPlugin {
+    constructor(private overrideStats: OverrideStatsEntry[]) {}
+
+    apply(compiler: Compiler) {
+        compiler.hooks.compilation.tap('SetOverrideStatsPlugin', (compilation: Compilation) => {
+            compilation.hooks.processAssets.tap(
+                {
+                    name: 'SetOverrideStats',
+                    stage: Compilation.PROCESS_ASSETS_STAGE_ANALYSE
+                },
+                () => {
+                    ;(compilation as any).overrideStats = this.overrideStats
+                }
+            )
+        })
+    }
+}
+
+describe('OverrideStatsPlugin', () => {
+    let originalRecordOverrides: string | undefined
+
+    // Preserve the original RECORD_OVERRIDES value and reset it after each test
+    beforeEach(() => {
+        originalRecordOverrides = process.env.RECORD_OVERRIDES
+    })
+
+    afterEach(() => {
+        if (originalRecordOverrides === undefined) {
+            delete process.env.RECORD_OVERRIDES
+        } else {
+            process.env.RECORD_OVERRIDES = originalRecordOverrides
+        }
+    })
+
+    const testCases: Array<
+        [
+            string,
+            {
+                recordOverrides: string
+                overrideStats: OverrideStatsEntry[]
+                expects: (output: any, fileSystem: any) => void
+            }
+        ]
+    > = [
+        [
+            'generates overrides-stats.json when RECORD_OVERRIDES is true',
+            {
+                recordOverrides: 'true',
```

Comment:
Why is the value a string?

Reply:
The value for `recordOverrides` is set via terminal environment variable `RECORD_OVERRIDES=true` and in Node.js, all values retrieved from process.env are strings.